app/routes.py

```diff
@@ -1,2 +1,264 @@
+def delete_video(video_id):
+    """Deletes a specific video."""
+    video = Video.query.get(video_id)
+    if not video:
+        return error_response(code=404)
+    db.session.delete(video)
+    db.session.commit()
+    return jsonify({"id": int(video_id)}), 200
+
+####################### RENTAL ROUTES #######################
+
+@rentals_bp.route("/check-out", methods = ["POST"])
+def check_out_video():
+    """Checks out a video to a customer, and updates the data in the database as such."""
+    request_body = request.get_json()
+    video_id = request_body.get("video_id")
+    customer_id = request_body.get("customer_id")
+
+    if type(customer_id) != int or type(video_id) != int:
+        return bad_request()
+
+    video = Video.query.get(video_id)
+    customer = Customer.query.get(customer_id)
+
+    if not customer or not video:
+        return not_found()
+    if video.available_inventory <= 0:
+        return bad_request()
+
+    rental = Rental(
+        customer_id=customer_id, 
+        video_id=video_id,
+        due_date = date.today() + timedelta(days=7)
+        )
+    
+    customer.videos_checked_out_count += 1
+    video.available_inventory -= 1
+    
+    db.session.add(rental)
+    db.session.commit()
+
+    return jsonify({
+        "customer_id": customer.customer_id,
+        "video_id": video.video_id,
+        "due_date": rental.due_date,
+        "videos_checked_out_count": customer.videos_checked_out_count,
+        "available_inventory": video.available_inventory
+    })
+
+@rentals_bp.route("/check-in", methods=["POST"])
+def check_in_video():
+    """Checks in a video to a customer, and updates the data in the database as such."""
+    request_body = request.get_json()
+    video_id = request_body.get("video_id")
+    customer_id = request_body.get("customer_id")
+
+    if type(customer_id) != int or type(video_id) != int:
+        return bad_request()
+
+    video = Video.query.get(video_id)
+    customer = Customer.query.get(customer_id)
+
+    if not customer or not video:
+        return not_found()
+
+    rental = Rental.query.filter_by(
+        customer_id=customer.customer_id,
+        video_id=video.video_id).one_or_none()
+
+    if rental is None:
+        return bad_request()   
+
+    video.available_inventory += 1
+    customer.videos_checked_out_count -= 1
+
+    db.session.delete(rental)
+    db.session.commit()
+
+    return jsonify({
+        "customer_id": customer_id,
+        "video_id": video_id,
+        "videos_checked_out_count": customer.videos_checked_out_count,
+        "available_inventory": video.available_inventory
+        }), 200
+
+@customers_bp.route("<customer_id>/rentals", methods=["GET"])
+def get_customer_rentals(customer_id):
+    """List the videos a customer currently has checked out"""
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return not_found()
+    
+    rentals = Rental.query.filter_by(customer_id=customer.customer_id)
```

Comment:
If you use the relationship you don't need to use the whole filter thingy.

You can instead do something like:

`rentals = customer.rentals`

Which helps get the database "junk" out of the routes file too.